statsd/statsd.go

```diff
@@ -121,30 +135,100 @@ func (c *Client) watch() {
 
 func (c *Client) append(cmd string) error {
 	c.Lock()
+	defer c.Unlock()
 	c.commands = append(c.commands, cmd)
 	// if we should flush, lets do it
 	if len(c.commands) == c.bufferLength {
 		if err := c.flush(); err != nil {
-			c.Unlock()
 			return err
 		}
 	}
-	c.Unlock()
 	return nil
 }
 
+func joinMaxSize(cmds []string, sep string, maxSize int) ([]string, []int) {
+	var buffer bytes.Buffer
+	buffer.Reset() //just to be sure...
+
+	frames := make([]string, 0)
+	ncmds := make([]int, 0)
+	sepLength := len(sep)
+
+	elem := 0
+	for _, cmd := range cmds {
+		var needed int
+
+		if elem == 0 {
+			needed = len(cmd)
+		} else {
+			needed = sepLength + len(cmd)
+		}
+
+		if buffer.Len()+needed <= maxSize {
+			if elem != 0 {
+				buffer.WriteString(sep)
+			}
+			buffer.WriteString(cmd)
+			elem++
+		} else if elem == 0 {
```

Comment:
We also should check that UDP packets do not get larger than 65467 bytes, beyond that is where metrics begin to get cut, but UDP packet is never dropped.
Also, if we decide to flush large commands regardless of size, maybe we should remove check on `sendMsg`